connectors/services/sync.py

```diff
@@ -134,10 +148,14 @@ async def _run(self):
                     logger.critical(e, exc_info=True)
                     self.raise_if_spurious(e)
                 finally:
-                    await syncs.join()
+                    await self.syncs.join()
                     if one_sync:
                         break
 
+                self.syncs = None
+                # Immediately break instead of sleeping
+                if not self.running:
+                    break
                 await self._sleeps.sleep(self.idling)
```

Comment:
```suggestion
                if self.running:
                    await self._sleeps.sleep(self.idling)
```

Reply:
It's possible, but it'd still need a `break` somewhere.

I intuitively feel that it's more convenient to have `break` wrapped in `if`, not the main logic.